apps/hash-graph/lib/graph/src/snapshot.rs

```diff
@@ -35,6 +37,63 @@ pub enum SnapshotEntry {
     Entity(EntitySnapshotRecord),
 }
 
+impl SnapshotEntry {
+    pub fn install_error_stack_hook() {
+        error_stack::Report::install_debug_hook::<Self>(|entry, context| match entry {
+            Self::Snapshot(global_metadata) => {
+                context.push_body(format!(
+                    "graph version: {}",
+                    global_metadata.block_protocol_module_versions.graph
+                ));
+            }
+            Self::DataType(data_type) => {
+                context.push_body(format!("data type: {}", data_type.metadata.record_id));
+                if context.alternate() {
+                    if let Ok(json) = serde_json::to_string_pretty(data_type) {
+                        context.push_appendix(format!("{}:\n{json}", data_type.metadata.record_id));
+                    }
+                }
+            }
+            Self::PropertyType(property_type) => {
+                context.push_body(format!(
+                    "property type: {}",
+                    property_type.metadata.record_id
+                ));
+                if context.alternate() {
+                    if let Ok(json) = serde_json::to_string_pretty(property_type) {
+                        context.push_appendix(format!(
+                            "{}:\n{json}",
+                            property_type.metadata.record_id
+                        ));
+                    }
+                }
+            }
+            Self::EntityType(entity_type) => {
+                context.push_body(format!("entity type: {}", entity_type.metadata.record_id));
+                if context.alternate() {
+                    if let Ok(json) = serde_json::to_string_pretty(entity_type) {
+                        context
+                            .push_appendix(format!("{}:\n{json}", entity_type.metadata.record_id));
+                    }
+                }
+            }
+            Self::Entity(entity) => {
+                let entity_id = format!(
+                    "{}%{}",
+                    entity.metadata.record_id.entity_id.owned_by_id,
+                    entity.metadata.record_id.entity_id.entity_uuid
+                );
+                context.push_body(format!("entity: {entity_id}"));
```

Comment:
Do you want to be calling two `format!`s in a row?

Reply:
I fixed this in [Implement Display for EntityId](https://github.com/hashintel/hash/pull/2357/commits/aa500094b05ee365ee5ad1e43a71f9812b6d9cae)